### PR TITLE
Fix Bug where 0 was returned as null/''

### PR DESCRIPTION
**Pull request by bethmaloney**
Currently we do a falsy check on the input/output of the TextFieldElement. As `0` is a falsy value this means that we return null or an empty string.

To fix this we check if the value is nullish instead.

You can test it by using the Number input control and inputting 0. Previously it would show a blank value but now it shows 0.

### CODE REVIEW

**Comment by bethmaloney** [comment]
Sorry need to rebuild storybook

**Comment by bethmaloney** [comment]
Rebuilt storybook. Should be good to go.

**Comment by akullpp** [comment]
Please release

File: packages/rhf-mui/src/TextFieldElement.tsx [hunk 1]
```diff
@@ -100,17 +100,16 @@ const TextFieldElement = forwardRef(function TextFieldElement<
         typeof transform?.input === 'function'
           ? transform.input
           : (value) => {
-              return value || ('' as TValue)
```
Comment (dohomi) [inline packages/rhf-mui/src/TextFieldElement.tsx:103]: I would change this to 
```ts
return value ?? ('' as TValue)
```

**Review by dohomi — COMMENTED** [review]
Could you have a look at the suggestion?

File: packages/rhf-mui/src/TextFieldElement.tsx [hunk 2]
```diff
@@ -100,17 +100,16 @@ const TextFieldElement = forwardRef(function TextFieldElement<
...
       output:
         typeof transform?.output === 'function'
           ? transform.output
           : (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
               const value = event.target.value
-              return (type === 'number' && value ? +value : value) as PathValue<
-                TFieldValues,
-                TName
-              >
+              return (
```
Comment (dohomi) [inline packages/rhf-mui/src/TextFieldElement.tsx:110]: I am afraid we have an edge case here:
if event.target.value is "" (empty string) then +"" == 0 thats why  I think we need to improve it in a different way

**Comment by dohomi** [comment]
@bethmaloney thanks for your time an contribution. I just got back from holiday and straight caught a bad cold thats why my response (and brain) still a bit slow. There are minor concerns but currently your changes would change an empty string into a "0" thats where we need to look at as first.
I merged your other PR's already they where 👍 thanks

**Comment by bethmaloney** [comment]
Hi @dohomi no problems. Let me fix that up

**Comment by bethmaloney** [comment]
Thinking about it a bit more I think it makes sense for a TextFieldElement with type='number' to return `Number | null | undefined`.

`null | undefined` would be returned when the field is blank while Number is returned when the field is not empty. The other option would be to return `Number | undefined | ''`. I feel that checking if the return value is undefined or a string would feel a little strange and might surprise consumers. Let me know what you think. Below is the behaviour of the changes I've just pushed (happy to change however you'd like).

No Default Value
image

Value added and cleared
image

0 value
image

Other value
image

File: packages/rhf-mui/src/TextFieldElement.tsx (hunk 1 above)
Comment (bethmaloney) [reply packages/rhf-mui/src/TextFieldElement.tsx:103]: done

**Pull request merged**